### ISSUE TITLE
Increase portability [meta-issue]

### ISSUE THREAD

**Issue by robrecord** [luccahuguet/yazelix#23]
Hi, would it be possible to separate the config that makes this work from the other config so that we can more easily incorporate into our own installations? Such as making the base config files includeable from our own larger config. Many thanks for your hard work on this.

**Comment by luccahuguet** [comment]
nice one, thank you for the idea

how is yazelix working for you? 

> thanks for your hard work on this

You're welcome! 

we'll create a portability strategy here and improve on that..

**Comment by luccahuguet** [comment]
I should note that both yazi's and zellij's configs wont touch your default configs

because they stay in a non-standard place for the configs

so that's a plus for safety and out-of-the-boxness

but it won't integrate into your personal config like you mention

**Comment by luccahuguet** [comment]
we should 
- [ ] understand what we want to do here in a general level: do yazi or zellij support importing configs?
- [ ] how to get there
- [ ] implement it

**Comment by luccahuguet** [comment]
Interesting, sounds like a good plan

yazelix is opinionated by nature, but if portability is increased, that would be a plus.

if we follow this approach, it would be probably be a nushell script.
One that generates the default config file for each program (zellij, nushell etc) then appends / overrides the yazelix stuff

I think we should split this issue into multiple ones, one for each program, because we might have to deal with this in a case-by-case scenario

yazi for instance has this: https://yazi-rs.github.io/docs/flavors/overview

so we'll turn this into a meta issue and split it up.

**Comment by luccahuguet** [comment]
- https://github.com/luccahuguet/yazelix/issues/29
- https://github.com/luccahuguet/yazelix/issues/28
- https://github.com/luccahuguet/yazelix/issues/27

**Comment by luccahuguet** [comment]
thank you Rob!

**Comment by luccahuguet** [comment]
I agree with you, what we want is something like what we have here in my nushell config files

https://github.com/luccahuguet/nu-files

at the end of the default config.nu and env.nu, I just append an import to my actual configs (called my_env.nu and my_config.nu)

too bad the nushell config is not really needed for yazelix haha

but it's a nice example of what we are looking for

**Comment by duttaoindril** [comment]
I was hoping for something with different shells, like `fish`

**Comment by luccahuguet** [comment]
> I was hoping for something with different shells, like `fish`

you can use your favorite shell! 

nushell is indeed a shell but it is also a programming language

in this project it is used exclusively as a programming language (instead of bash scripting)

so you could run it on fish, so long as you have nushell in your system too

**Comment by luccahuguet** [comment]
> I was hoping for something with different shells, like `fish`

@duttaoindril , now you can set fish as your default shell in yazelix.nix

**Comment by luccahuguet** [comment]
@robrecord we are fixing yazi and zellij portability for good, for v8.5/v9

**Comment by luccahuguet** [comment]
we will have gitignored user configs

that get dynamically merged with yazelix's own
